openmc/tally_derivative.py

```diff
@@ -0,0 +1,168 @@
+        self._nuclide = None
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy this object, create a
+        # copy
+        if existing is None:
+            clone = type(self).__new__(type(self))
+            clone.id = self.id
+            clone.variable = self.variable
+            clone.material = self.material
+            clone.nuclide = self.nuclide
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    def __eq__(self, other):
+        if not isinstance(other, TallyDerivative):
+            return False
+
+        return (self.variable == other.variable and
+                self.material == other.material and
+                self.nuclide == other.nuclide)
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __hash__(self):
+        return hash(repr(self))
+
+    def __repr__(self):
+        string = 'Tally Derivative\n'
+        string += '{0: <16}{1}{2}\n'.format('\tID', '=\t', self.id)
+        string += '{0: <16}{1}{2}\n'.format('\tVariable', '=\t',
+                                            self.variable)
+
+        if self.variable == 'density':
+            string += '{0: <16}{1}{2}\n'.format('\tMaterial', '=\t',
+                                                self.material)
+        elif self.variable == 'nuclide_density':
+            string += '{0: <16}{1}{2}\n'.format('\tMaterial', '=\t',
+                                                self.material)
+            string += '{0: <16}{1}{2}\n'.format('\tNuclide', '=\t',
+                                                self.nuclide)
+        elif self.variable == 'temperature':
+            string += '{0: <16}{1}{2}\n'.format('\tMaterial', '=\t',
+                                                self.material)
+
+        return string
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def variable(self):
+        return self._variable
+
+    @property
+    def material(self):
+        return self._material
+
+    @property
+    def nuclide(self):
+        return self._nuclide
+
+    @id.setter
+    def id(self, deriv_id):
+        if deriv_id is None:
+            global AUTO_TALLY_DERIV_ID
+            self._id = AUTO_TALLY_DERIV_ID
+            AUTO_TALLY_DERIV_ID += 1
+        else:
+            cv.check_type('tally derivative ID', deriv_id, Integral)
+            cv.check_greater_than('tally derivative ID', deriv_id, 0,
+                                  equality=True)
+            self._id = deriv_id
+
+    @variable.setter
+    def variable(self, var):
+        if var is not None:
+            cv.check_type('derivative variable', var, string_types)
+            cv.check_value('derivative variable', var, ('density',
+                           'nuclide_density', 'temperature'))
```

Comment:
line continuation context doesn't really match -- it would lead one to believe that 'nuclide_density' is part of the same argument tuple as 'derivative variable' above it. Should really be:
```Python
cv.check_value('derivative variable', var,
               ('density', 'nuclide_density', 'temperature'))
```